scheduler/tasks.go

```diff
@@ -0,0 +1,361 @@
+package scheduler
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/PlakarKorp/plakar/appcontext"
+	"github.com/PlakarKorp/plakar/cmd/plakar/subcommands/backup"
+	"github.com/PlakarKorp/plakar/cmd/plakar/subcommands/check"
+	"github.com/PlakarKorp/plakar/cmd/plakar/subcommands/cleanup"
+	"github.com/PlakarKorp/plakar/cmd/plakar/subcommands/restore"
+	"github.com/PlakarKorp/plakar/cmd/plakar/subcommands/rm"
+	"github.com/PlakarKorp/plakar/cmd/plakar/subcommands/sync"
+	"github.com/PlakarKorp/plakar/repository"
+	"github.com/PlakarKorp/plakar/storage"
+)
+
+func (s *Scheduler) backupTask(taskset TaskSet, task BackupConfig) error {
+	interval, err := stringToDuration(task.Interval)
+	if err != nil {
+		return err
+	}
+
+	var retention time.Duration
+	if task.Retention != "" {
+		retention, err = stringToDuration(task.Retention)
+		if err != nil {
+			return err
+		}
+	}
+
+	backupSubcommand := &backup.Backup{}
+	backupSubcommand.RepositoryLocation = taskset.Repository.URL
+	if taskset.Repository.Passphrase != "" {
+		backupSubcommand.RepositorySecret = []byte(taskset.Repository.Passphrase)
+		_ = backupSubcommand.RepositorySecret
```

Comment:
debugging leftover?

Reply:
placeholder for however we're going to tackle this :-/